src/mongocxx/write_concern.hpp

```diff
@@ -55,6 +55,15 @@ class bulk_write;
 class MONGOCXX_API write_concern {
 
    public:
+    ///
+    /// A class to represent the special case values for write_concern::nodes.
+    /// @see http://docs.mongodb.org/manual/reference/write-concern/#w-option
+    ///
+    enum class level : std::int32_t {
```

Comment:
Do we actually want to expose the numeric values here? This is sort of leaking an implementation detail. It might be better to make these symbolic and map. Also, per my other comment, you might need to represent kUnknown.